File: api_nestjs_distributed_order_platform/src/application/port/order.repository.ts
import { OrderEntity } from "src/domain/entities/order.entity";
import { Transaction } from "sequelize";

export interface OrderRepositoryInterface {
    create(order: OrderEntity, transaction?: Transaction): Promise<OrderEntity>;
    findById(orderId: string): Promise<OrderEntity | null>;
    findAll(): Promise<OrderEntity[]>;
    update(order: OrderEntity, transaction?: Transaction): Promise<void>;
    delete(orderId: string): Promise<void>;
}

File: api_java_distributed_order_platform/src/main/java/com/project/order/application/port/OrderRepository.java
package com.project.order.application.port;

import com.project.order.domain.model.Order;

import java.util.List;
import java.util.Optional;

public interface OrderRepository {
    Order create(Order order);
    Optional<Order> findById(String id);
    List<Order> findAll();
    Order update(Order order);
    void delete(String id);
}
